torch/_dynamo/output_graph.py

```diff
@@ -595,21 +595,30 @@ def save_global_state(self, out=None):
             self.torch_function_enabled,
         )
         global_state["grad_enabled"] = (torch.set_grad_enabled, torch.is_grad_enabled())
+
+        def autocast_specific_backend(
+            device_type: str, func: Callable[[str, Any], None]
+        ):
+            def decorator(value):
+                return func(device_type, value)
+
+            return decorator
+
         global_state["autocast_enabled"] = (
-            torch.set_autocast_enabled,
-            torch.is_autocast_enabled(),
+            autocast_specific_backend("cuda", torch.set_autocast_enabled),
```

Comment:
Some dynamo-related UTs can catch warnings as errors. So it results in UT failures if we use deprecated APIs here.